dask/array/tests/test_reductions.py

```diff
@@ -809,3 +809,28 @@ def test_nan_func_does_not_warn(func):
     with pytest.warns(None) as rec:
         getattr(da, func)(d).compute()
     assert not rec  # did not warn
+
+
+@pytest.mark.parametrize("chunks", list(permutations(((2, 1) * 8, (3,) * 8, (6,) * 4))))
+@pytest.mark.parametrize("split_every", [2, 4, 8, 16])
+@pytest.mark.parametrize(
+    "axes", list(permutations((0, 1, 2), 2)) + list(permutations((0, 1, 2)))
+)
```

Comment:
If I did the math correctly, this is 288 tests. If they're fast, then this may be fine (although they'll clutter up the test output a bit). If they're slow at all then this is undesirable and we may want to reduce the parametrization space a bit. Up to you.

Reply:
You're right &mdash; it's 288, and they take one and a half minutes in total to run on my laptop.  Is that ok?  I felt it was important to test all the possible cases, especially after the original test I wrote proved insufficient to capture all possible bugs.

Reply:
Yeah, we'll want to cut this down significantly then. For reference, the whole test suite runs in 20-30 min on CI (your laptop is likely much faster) - taking > 5% of that time to check one method seems excessive.

Since you've thought about this problem more than I have, can you to think about what cases check what parts of the code? Perhaps you don't need `chunks * split_every * axes` different test cases, perhaps selecting a few distinct subsets would be sufficient? Or maybe cutting back on the number of `axes` & `split_every` cases that are checked? I'd accept these tests running in 30 seconds (1/3 the current time), but would be happier if they took closer to 15 seconds. Up to you how to accomplish that.

Reply:
OK.  I've reduced it to 144 cases which take 23 seconds in all.  I only reduced `split_every`'s choices to `[2, 4]`